src/com/sap/piper/Notify.groovy

```diff
@@ -0,0 +1,51 @@
+package com.sap.piper
+
+import groovy.text.SimpleTemplateEngine
+
+class Notify implements Serializable {
+    private static enum Severity { ERROR, WARNING }
+    private final static String LIBRARY_NAME = 'piper-lib-os'
+    private final static String MESSAGE_PATTERN = '[${severity}] ${message} (${libName}/${stepName})'
+
+    protected static Utils instance = null
+
+    protected static Utils getUtilsInstance(){
+        instance = instance ?: new Utils()
+        return instance
+    }
+
+    static void warning(Map config, Script step, String msg, String stepName = null){
+        log(config, step, msg, stepName)
+    }
+
+    static void error(Map config, Script step, String msg, String stepName = null) {
+        log(config, step, msg, stepName, Severity.ERROR)
+    }
+
+    private static void log(Map config, Script step, String msg, String stepName, Severity severity = Severity.WARNING){
```

Comment:
Would be more clear IMO if there is no default severity, but the severity is always added in the corresponding method (`error`, `warning`, maybe `info`, `debug` in the future.

Reply:
removed default